### PR TITLE
Fix A5 import logic

### PR DESCRIPTION
**Pull request by prognant**
### What does this PR do? 
Fix A5 import sequence since a major issue was introduced by #4813.

Roughly half the configuration was reset in the middle of the conversion process. Thus  at the end of the conversion process, some parameter were left un-imported (and back to their default A6 value) and the imported configuration was then missing significant settings, most notably `api_key`.

Some integration path were imported into the wrong location, now `conf.d/docker.d/conf.yaml` and `conf.d/kubelet.d/conf.yaml` are properly created.

End-to-end import test have been added, the resulting file are read & unmarshalled directly without going through the configuration pipeline to prevent default value interfering with the actual test.

### Motivation
Fix A5 import logic, prevent future occurence of issue like the one introduced by #4813.

### Additional Notes
N/A

### Describe your test plan
Manual test + new end-to-end import test.

### QA consideration
To be checked: 
* The import generates a valid a6/7 
* Critical parameters importation (api key, dd_url)
* Correct file/dir layout in imported conf (conf.d/<integration>.d/conf.yaml)
* With a full blown working A5 configuration ([datadog.conf](https://github.com/DataDog/dd-agent/blob/master/datadog.conf.example) + some integrations) checking that A6 is fully functional, ideally using the official upgrade path to A6 (https://docs.datadoghq.com/agent/versions/upgrade_to_agent_v6/?tab=linux)

### CODE REVIEW

File: releasenotes/notes/fix-broken-a5-import-a2358410cf4a37da.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Agent 5 configuration import was missing critical parameters.
```
Comment (olivielpeau) [inline releasenotes/notes/fix-broken-a5-import-a2358410cf4a37da.yaml:11]: what was the user impact? Would be nice to explain this in more details if possible

**Review by olivielpeau — COMMENTED** [review]
Thanks for adding all these tests 👍 

Could you describe in the PR description the major issue that this PR fixes? I don't really understand what the actual issue is, and how this fixes it.

Also, could you add more details on the things we should test during QA?

According to https://github.com/DataDog/datadog-agent/pull/2559 the goal of the `LegacyConfigConverter` was to isolate uses of `Config.Set`, so that we could ultimately make `Set` private. We didn't actually finish this, but I don't know if we should completely abandon this. Maybe keeping it and adding more documentation would be enough, WDYT?

File: releasenotes/notes/fix-broken-a5-import-a2358410cf4a37da.yaml (hunk 1 above)
Comment (prognant) [reply releasenotes/notes/fix-broken-a5-import-a2358410cf4a37da.yaml:11]: Having `config.Set` private would indeed be a valuable move. I will keep the wrapping logic in that extend. I updated the PR description with a more detailed explanation.

**Comment by prognant** [comment]
I brought back the legacy config converter wrapper & improved the release note with additional details.

File: releasenotes/notes/fix-broken-a5-import-a2358410cf4a37da.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    The configuration was reset to default settings in the middle
+    of the migration process. It was leading to many unimported
+    parameters including very important ones like ``api_key``.
+    Some Docker & Kubernetes configuration settings were also
+    imported in wrongly located files.
```
Comment (olivielpeau) [inline releasenotes/notes/fix-broken-a5-import-a2358410cf4a37da.yaml:16]: nit: for conciseness:
```suggestion
    Fix bugs introduced in 7.20.0/6.20.0 in the Agent 5 configuration import command:
    the command would not import some Agent config settings, including ``api_key``,
    and would write some Docker & Kubernetes config settings to wrongly-located files.
```

**Review by olivielpeau — APPROVED** [review]
LGTM, thanks! Made a suggestion on the release note.

Can you add details on the scenarios that we should validate during QA? I think it'll be useful during QA.

Also, just for completeness/FYI: `conf.d/docker.yaml` is a valid integration config paths for Agent 6/7, even though the "standard" now is to use `conf.d/docker.d/<some_filename>.yaml` instead.

**Pull request merged**